flag_engine/context/mappers.py

```diff
@@ -19,6 +31,77 @@ def map_environment_identity_to_context(
     :param override_traits: A list of TraitModel objects, to be used in place of `identity.identity_traits` if provided.
     :return: An EvaluationContext containing the environment and identity.
     """
+    features = map_feature_states_to_feature_contexts(environment.feature_states)
+    segments: typing.Dict[str, SegmentContext] = {}
+    for segment in environment.project.segments:
+        segment_ctx_data: SegmentContext = {
+            "key": str(segment.id),
+            "name": segment.name,
+            "rules": map_segment_rules_to_segment_context_rules(segment.rules),
+        }
+        if segment_feature_states := segment.feature_states:
+            segment_ctx_data["overrides"] = list(
+                map_feature_states_to_feature_contexts(segment_feature_states).values()
+            )
+        segments[segment.name] = segment_ctx_data
+    # Concatenate feature states overriden for identities
+    # to segment contexts
+    features_to_identifiers: typing.Dict[
+        typing.Tuple[typing.Tuple[str, str, bool, typing.Any], ...],
+        typing.List[str],
+    ] = {}
```

Comment:
Do you think a _typedef_ could improve readability? Also, `defaultdict` suggested.

```suggestion
    features_to_identifiers: defaultdict[FeatureOverrideKey, list[str]] = defaultdict(list)
```

Also, and out of scope, I think we should update [Python compatibility](https://github.com/Flagsmith/flagsmith-engine/blob/bef422ce7053d3e66ad9f1ad43ea61ca642583a4/setup.py#L23-L25). Among other things, we could benefit from type hinting with e.g. `dict` and `tuple` instead of `typing.Dict` and `typing.Tuple`, respectively.